src/SArray.jl

```diff
@@ -50,6 +50,28 @@ end
     end
 end
 
+"""
+    SArray{S, T, N, L}(undef)
+    SArray{S, T, N}(undef)
+    SArray{S, T}(undef)
+"""
+@inline SArray{S, T, N, L}(::UndefInitializer) where {S <: Tuple, T, N, L} = 
+    SArray{S, T, N, L}(Array{T}(undef, size_to_tuple(S) ) )
```

Comment:
This allocates a normal `Array`, and then discards it which would be a nasty performance trap. If we must supply this constructor, you should use an `MArray` here.

Reply:
I would like to use SVector as one of the fields in a **mutable** struct. So, the whole mutable object has to be constructed in the first place (by creating a `undef SVector`), and then we could mutate the field of it.